### PR TITLE
Vaccine request/config product vaccine

### PR DESCRIPTION
**Pull request by rindibudiaramdhan**
Add default conditional category when access Product table.
If request has parameter category, then query by that category, else category will lock at "ALKES" and "NON ALKES" only.

This condition added because product list of vaccine feature should show only at vaccine feature only.

@jabardigitalservice/jds-backend 

## Evidence
title: Vaccine request/config product vaccine #317
project: Pikobar Logistik
participants: @budiaramdhanrindi @sandisunandar @firmanalamsyah580

### CODE REVIEW

**Review by sandisunandar99 — APPROVED** [review]

File: app/Http/Controllers/API/v1/ProductsController.php [hunk 1]
```diff
@@ -108,6 +121,13 @@ public function productTopRequest(Request $request)
                                     ->withCount(['need as total' => function($query) use ($request) {
                                         $query->select(DB::raw('sum(quantity)'))->filterByApplicant($request);
                                     }])
+                                    ->where(function ($query) use ($request) {
+                                        if ($request->has('category')) {
+                                            $query->where('category', $request->input('category'));
+                                        } else {
+                                            $query->where('category', '!=', 'VAKSIN');
```
Comment (ayocodingit) [inline app/Http/Controllers/API/v1/ProductsController.php:128]: Can use Enum class 😁

File: app/Http/Controllers/API/v1/ProductsController.php [hunk 2]
```diff
@@ -108,6 +121,13 @@ public function productTopRequest(Request $request)
                                     ->withCount(['need as total' => function($query) use ($request) {
                                         $query->select(DB::raw('sum(quantity)'))->filterByApplicant($request);
                                     }])
+                                    ->where(function ($query) use ($request) {
```
Comment (ayocodingit) [inline app/Http/Controllers/API/v1/ProductsController.php:124]: This duplicate code. Can use scope query ?

**Review by ayocodingit — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/API/v1/ProductsController.php [hunk 3]
```diff
@@ -78,6 +84,13 @@ public function productRequest(Request $request)
                 ->withCount(['need as total_request' => function($query) use ($request) {
                     $query->select(DB::raw('sum(quantity)'))->filterByApplicant($request);
                 }])
+                ->where(function ($query) use ($request) {
+                    if ($request->has('category')) {
```
Comment (sandisunandar99) [inline app/Http/Controllers/API/v1/ProductsController.php:88]: oia ,, ga kena code climate duplication ini? wkwkw

Comment (ayocodingit) [reply app/Http/Controllers/API/v1/ProductsController.php:88]: Wkwk codeclimate gk strict di php 😅

File: app/Http/Controllers/API/v1/ProductsController.php (hunk 1 above)
Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/ProductsController.php]: done! 😁

File: app/Http/Controllers/API/v1/ProductsController.php (hunk 2 above)
Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/ProductsController.php]: done! 😁

File: app/Http/Controllers/API/v1/ProductsController.php (hunk 3 above)
Comment (rindibudiaramdhan) [reply app/Http/Controllers/API/v1/ProductsController.php]: kena mang. done! 😁 saya buat scope function

**Review by ayocodingit — APPROVED** [review]
Good Job @rindibudiaramdhan :+1:

**Pull request merged**